circuitx/navigation/src/commonMain/kotlin/com/slack/circuitx/navigation/intercepting/CircuitInterceptingNavigator.kt

```diff
@@ -0,0 +1,199 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuitx.navigation.intercepting
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.SideEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.slack.circuit.foundation.internal.BackHandler
+import com.slack.circuit.retained.rememberRetained
+import com.slack.circuit.runtime.Navigator
+import com.slack.circuit.runtime.screen.PopResult
+import com.slack.circuit.runtime.screen.Screen
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+
+/**
+ * Creates and remembers a [CircuitNavigationInterceptor] from a [Navigator].
+ *
+ * @param navigator The [Navigator] to delegate to.
+ * @param interceptors The [CircuitNavigationInterceptor]s to intercept navigation events.
+ * @param eventListeners The [CircuitNavigationEventListener]s to notify.
+ * @param notifier An optional [CircuitInterceptingNavigator.FailureNotifier] to notify of
+ *   [CircuitNavigationInterceptor]
+ * @param enableBackHandler Whether to enable the back button handler. When enabled back handling
+ *   will be triggered on the [CircuitInterceptingNavigator] as a [Navigator.pop] to be intercepted.
+ * @see Navigator
+ * @see CircuitInterceptingNavigator
+ */
+@Composable
+public fun rememberCircuitInterceptingNavigator(
+  navigator: Navigator,
+  interceptors: ImmutableList<CircuitNavigationInterceptor> = persistentListOf(),
+  eventListeners: ImmutableList<CircuitNavigationEventListener> = persistentListOf(),
+  notifier: CircuitInterceptingNavigator.FailureNotifier? = null,
+  enableBackHandler: Boolean = true,
+): Navigator {
+  // Handle the NavigationInterceptors.
+  val interceptingNavigator =
+    remember(navigator, interceptors, eventListeners, notifier) {
+      CircuitInterceptingNavigator(
+        delegate = navigator,
+        interceptors = interceptors,
+        eventListeners = eventListeners,
+        notifier = notifier,
+      )
+    }
+  // Handle the back button here to get pop events from it.
+  if (enableBackHandler) {
+    var trigger by remember { mutableStateOf(false) }
+    BackHandler(!trigger && enableBackHandler) {
```

Comment:
Will `enableBackHandler` always be true here?

Reply:
Yup! I'll clean that up